### PR TITLE
Tutorials

### PR DESCRIPTION
**Pull request by alexandra2607**
### Requirements

The PR adds suport for tutorials.

### Format

[x] ran `npm run electron-format`
[x] ran `npm run browser-format`

### Author
Signed-off-by: Your Full Name <youremailaddress@emailprovider.domain>

### CODE REVIEW

File: source/plugins/projects/projects/index.js [hunk 1]
```diff
@@ -491,7 +491,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * importProject('MyNewProject', '.zip');
+	 * importProject(project, '.zip');
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:494]: This does not seem right, `importProject` takes 3 parameters while in the documentation there are only 2.

File: source/plugins/projects/projects/index.js [hunk 2]
```diff
@@ -777,7 +777,7 @@ let projects = {
 	 *
 	 * @example
 	 * 
-	 * newFolder('MyNewProject', 'C:\Users\User\Desktop');
+	 * newFolder(project, 'C:\Users\User\Desktop');
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:780]: The second parameter is a folder within the project, it cannot be with a full path, it has to start with /...

File: source/plugins/projects/projects/index.js [hunk 3]
```diff
@@ -973,7 +973,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * deleteFile('MyNewProject', 'C:\Users\User\Desktop\file');
+	 * deleteFile(project, 'C:\Users\User\Desktop\file');
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:976]: The second parameter is a folder within the project, it cannot be with a full path, it has to start with /...

File: source/plugins/projects/projects/index.js [hunk 4]
```diff
@@ -1023,7 +1023,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * deleteFolder('MyNewProject', 'C:\Users\User\Desktop\folder');
+	 * deleteFolder(project, 'C:\Users\User\Desktop\folder');
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:1026]: The second parameter is a folder within the project, it cannot be with a full path, it has to start with /...

File: source/plugins/projects/projects/index.js [hunk 5]
```diff
@@ -1217,7 +1217,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * 		saveFile('MyNewProject', 'FileName', [1, 2, 3]);
+	 * 		saveFile(project, 'FileName', [1, 2, 3]);
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:1220]: As most of the data written in a file will be strings, I suggest having the example with `Buffer.from ('.....')`

File: source/plugins/projects/projects/index.js [hunk 6]
```diff
@@ -1329,7 +1329,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * saveSpecialFile('MyNewProject', 'SpecialFileName', [1, 2, 3]);
+	 * saveSpecialFile(project, 'SpecialFileName', [1, 2, 3]);
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:1332]: As most of the data written in a file will be strings, I suggest having the example with Buffer.from ('.....')

File: source/plugins/projects/projects/index.js [hunk 7]
```diff
@@ -1607,7 +1607,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * let sourceLanguage = languageSpecificOption ('MyNewProject', 'sourceLanguage');
+	 * let sourceLanguage = languageSpecificOption (project, 'sourceLanguage');
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:1610]: options is an object

File: source/plugins/tutorials/index.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+			tutorials.showTutorials ('alexandruradovici/tutorials-wyliodrin');
+		},
+
+		'plugins/tutorials/data/img/toque.png');
+		
+	register(null, {
+		tutorials: tutorials
+	});
+	
+}
```
Comment (alexandruradovici) [inline source/plugins/tutorials/index.js:34]:
```suggestion
}

```

File: source/plugins/tutorials/package.json [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+	"name": "tutorials",
+	"version": "0.0.1",
+	"main": "index.js",
+	"private": false,
+	"plugin": {
+		"consumes": ["workspace", "projects"],
+		"provides": ["tutorials"],
+		"target": ["browser", "electron"]
+	}
+}
```
Comment (alexandruradovici) [inline source/plugins/tutorials/package.json:11]:
```suggestion
}

```

File: source/plugins/tutorials/translations/messages-en.json [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+	},
+	"TUTORIALS_IMPORT_PROJECT_NAME": {
+		"message": "Input the name for the new project",
+		"description": "Name of the new project"
+	},
+	"TUTORIALS_PROJECT_EXISTS": {
+		"message": "A project with the name {name} already exists, please choose another name for it.",
+		"description": "A message shown if there is already a project with that name"
+	}
+}
```
Comment (alexandruradovici) [inline source/plugins/tutorials/translations/messages-en.json:18]:
```suggestion
}

```

File: source/plugins/tutorials/translations/messages-fr.json [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+	},
+	"TUTORIALS_IMPORT_PROJECT_NAME": {
+		"message": "Mettez le nom du nouveau projet.",
+		"description": "Le nom du nouveau projet."
+	},
+	"TUTORIALS_PROJECT_EXISTS": {
+		"message": "Un projet avec le nom {name} existe déjà, s'il vous plaît choisissez un autre nom",
+		"description": "Un message apparaît si il existe déjà un projet avec le même nom"
+	}
+}
```
Comment (alexandruradovici) [inline source/plugins/tutorials/translations/messages-fr.json:18]:
```suggestion
}

```

File: source/plugins/tutorials/translations/messages-ro.json [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+	},
+	"TUTORIALS_IMPORT_PROJECT_NAME": {
+		"message": "Scrieţi numele noului proiect",
+		"description": "Numele noului proiect"
+	},
+	"TUTORIALS_PROJECT_EXISTS": {
+		"message": "Un proiect cu numele {name} există deja, vă rog alegeţi un alt nume.",
+		"description": "Apare un mesaj dacă există deja un proiect cu același nume"
+	}
+}
```
Comment (alexandruradovici) [inline source/plugins/tutorials/translations/messages-ro.json:18]:
```suggestion
}

```

File: source/plugins/tutorials/views/Tutorials.vue [hunk 13]
```diff
@@ -0,0 +1,182 @@
...
+				}
+			}	
+		},
+		async downloadFile (path) {
+			let file = await axios.get (`https://raw.githubusercontent.com/${this.repository}/master/${path}`, {responseType: 'arraybuffer',});
+			return file.data;
+		}
+	}
+};
+</script>
```
Comment (alexandruradovici) [inline source/plugins/tutorials/views/Tutorials.vue:182]:
```suggestion
</script>

```

**Review by alexandruradovici — COMMENTED** [review]
Delete ` .metals/metals.h2.db`

File: docs/source/boards/esp8266.rst [hunk 14]
```diff
@@ -0,0 +1,201 @@
+=================================
+Install and configure Micropython
```
Comment (alexandruradovici) [inline docs/source/boards/esp8266.rst:2]:
```suggestion
ESP 8266 (MicroPython)
```

File: source/plugins/projects/projects/index.js (hunk 3 above)
Comment (alexandruradovici) [reply source/plugins/projects/projects/index.js]: The path still starts with `C:\...`

File: source/plugins/projects/projects/index.js (hunk 4 above)
Comment (alexandruradovici) [reply source/plugins/projects/projects/index.js:1031]: The path still starts with `C:\...`

File: source/plugins/projects/projects/index.js [hunk 15]
```diff
@@ -782,7 +782,7 @@ let projects = {
 	 *
 	 * @example
 	 * 
-	 * newFolder('MyNewProject', 'C:\Users\User\Desktop');
+	 * newFolder(project, '.folder/file');
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:785]:
```suggestion
	 * newFolder(project, '/folder/folder2');
```

File: source/plugins/projects/projects/index.js [hunk 16]
```diff
@@ -927,7 +927,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * renameObject('MyNewProject', 'ObjectNewName', 'C:\Users\User\Desktop');
+	 * renameObject(project, 'ObjectNewName', 'C:\Users\User\Desktop');
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:930]: Same as folder, use a relative path

File: source/plugins/projects/projects/index.js [hunk 17]
```diff
@@ -978,7 +978,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * deleteFile('MyNewProject', 'C:\Users\User\Desktop\file');
+	 * deleteFile(project, '.folder/file');
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:981]:
```suggestion
	 * deleteFile(project, '/folder/file');
```

File: source/plugins/projects/projects/index.js (hunk 4 above)
Comment (alexandruradovici) [reply source/plugins/projects/projects/index.js:1031]: Same as deleteFile

File: source/plugins/projects/projects/index.js [hunk 18]
```diff
@@ -1222,7 +1222,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * 		saveFile('MyNewProject', 'FileName', [1, 2, 3]);
+	 * 		saveFile(project, 'FileName', [1, 2, 3]);
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:1225]:
```suggestion
	 * 		saveFile(project, '/folder/file', Buffer.from ('...'));
```

File: source/plugins/projects/projects/index.js [hunk 19]
```diff
@@ -1334,7 +1334,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * saveSpecialFile('MyNewProject', 'SpecialFileName', [1, 2, 3]);
+	 * saveSpecialFile(project, 'SpecialFileName', [1, 2, 3]);
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:1337]:
```suggestion
	 * saveSpecialFile(project, 'SpecialFileName', Buffer.from ('...'));
```

File: source/plugins/projects/projects/index.js [hunk 20]
```diff
@@ -1371,7 +1371,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * deleteSpecialFile('MyNewProject', 'SpecialFileName');
+	 * loadSpecialFile(project, 'SpecialFileName');
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:1374]:
```suggestion
	 * deleteSpecialFile(project, 'SpecialFileName');
```

File: source/plugins/projects/projects/index.js [hunk 21]
```diff
@@ -1587,7 +1587,7 @@ let projects = {
 	 * 
 	 * @example
 	 * 
-	 * let sourceLanguage = languageSpecificOption ('MyNewProject', 'sourceLanguage');
+	 * let sourceLanguage = languageSpecificOption (project, 'sourceLanguage');
```
Comment (alexandruradovici) [inline source/plugins/projects/projects/index.js:1590]:
```suggestion
	 * let sourceLanguage = languageSpecificOption (project, {...});
```

File: docs/source/boards/esp8266.rst [hunk 22]
```diff
@@ -0,0 +1,201 @@
...
+
+	**$ screen /dev/ttyUSB0 115200**
+
+“115200” is the speed read by serial port. To close it type **Ctrl+D** or **Ctrl+a** followed by **Ctrl+\\**.
+
+**Congratulations**, now you have Micropython on your ESP8266!
+
+|
+
+3. Install and configure Micropython on MacOS
```
Comment (alexandruradovici) [inline docs/source/boards/esp8266.rst:164]:
```suggestion
macOS
*********
```

File: docs/source/boards/esp8266.rst [hunk 23]
```diff
@@ -0,0 +1,201 @@
...
+---------------------------------------------
+
+3.1 Download Micropython firmware
```
Comment (alexandruradovici) [inline docs/source/boards/esp8266.rst:167]:
```suggestion
1. Download Micropython firmware
```

File: docs/source/boards/esp8266.rst [hunk 24]
```diff
@@ -0,0 +1,201 @@
...
+
+Open a browser and type this link: https://micropython.org/download/esp8266/,
+then go install the latest version (without opening it). 
+
+*Suggestion*: create a folder on your computer named “esp8266” or “micropython” and download it there.
+
+.. image:: images/esp8266/mp_firmware.png
+	:align: center
+
+3.2 Install NodeMCU PyFlasher and flashing Micropython on ESP8266
```
Comment (alexandruradovici) [inline docs/source/boards/esp8266.rst:178]:
```suggestion
2. Install NodeMCU PyFlasher and flashing Micropython on ESP8266
```

File: docs/source/boards/esp8266.rst [hunk 25]
```diff
@@ -0,0 +1,201 @@
...
+First, you have *to connect the ESP8266 to your computer*. Take the USB cable from the kit and Put the USB-C in one of your ports and then connect the micro-USB to the microcontroller.
+
+Check this link `install NodeMCU <https://github.com/marcelstoer/nodemcu-pyflasher/releases>`_, if you have High Sierra. Scroll down to the executables and click on the third executable (*NodeMCU-PyFlasher-4.0.dmg*).
+
+After the installation will be completed, click on the NodeMCU PyFlasher .exe file and you should have a similar window: 
+
+.. image:: images/esp8266/nodeMCU_mac.png
+	:align: center
+
+First, you have to choose your serial port, in this case the port is: “/dev/cu.SLAB_USBtoUART”
```
Comment (alexandruradovici) [inline docs/source/boards/esp8266.rst:191]: Yopu might be missing a subtitle here like "Flashing Micropython"?

Comment (alexandra2607) [reply docs/source/boards/esp8266.rst:193]: It's the same subtitle structure as Windows "Install NodeMCU PyFlasher and flashing Micropython on ESP8266". 
Should I divide them like "Install NodeMCU PyFlasher" and " Flashing Micropython on ESP8266" ?

Comment (alexandruradovici) [reply docs/source/boards/esp8266.rst:193]: Got it

**Review by alexandruradovici — APPROVED** [review]

**Pull request merged**